props/saber_BC_buttons.h

```diff
@@ -477,738 +1492,1266 @@ class SaberBCButtons : public PROP_INHERIT_PREFIX PropBase {
+        return true;
+#endif
+
+// Revert ColorChange to uploaded config color (reset Variation)
+      case EVENTID(BUTTON_POWER, EVENT_THIRD_CLICK_LONG, MODE_ON):
+        if (spam_blast_) return false;
+        PVLOG_NORMAL << "** Reverted Color Variation to uploaded config color. Variation = " << SaberBase::GetCurrentVariation() << "\n";
+        SaberBase::SetVariation(0);
+        if (SFX_mnum) {
+          sound_library_v2.SayResetToDefaultColor();
+        } else {
+          beeper.Beep(0.20, 2000.0);
+          beeper.Beep(0.20, 1414.2);
+          beeper.Beep(0.20, 1000.0);
+        }
+        return true;
+
+// Quote - NOT pointing DOWN
+// Toggle Sequential/Random quotes - pointing DOWN
+      case EVENTID(BUTTON_POWER, EVENT_THIRD_SAVED_CLICK_SHORT, MODE_ON):
+      case EVENTID(BUTTON_POWER, EVENT_THIRD_SAVED_CLICK_SHORT, MODE_OFF):
+        if (fusor.angle1() < -M_PI / 4) {
+          // pointing DOWN
+          ToggleSequentialQuote();
+        } else {
+          DoQuote();
+        }
+        return true;
+
+// User Effects a.k.a. "Special Abilities" ©Fett263
+      case EVENTID(BUTTON_NONE, EVENT_TWIST_LEFT, MODE_ON | BUTTON_POWER):
+        if (!scroll_presets_) {
+          PVLOG_NORMAL << "** EFFECT_USER1 **\n";
+          SaberBase::DoEffect(EFFECT_USER1, 0);
           return true;
         }
-#endif
-        // pointing up
-        if (fusor.angle1() >  M_PI / 3) {
+        current_menu_angle_ = (fusor.angle2() - (M_PI / 6));
+        return false;
+
+      case EVENTID(BUTTON_NONE, EVENT_TWIST_RIGHT, MODE_ON | BUTTON_POWER):
+        if (!scroll_presets_) {
+          PVLOG_NORMAL << "** EFFECT_USER2 **\n";
           SaberBase::DoEffect(EFFECT_USER2, 0);
-        } else {
-         // NOT pointing up OR down
-          SaberBase::DoForce();
+          return true;
         }
-      }
-      return true;
-
-// Quote
-// Revert Color Change without saving (reset to Variation == 0)
-    case EVENTID(BUTTON_POWER, EVENT_THIRD_SAVED_CLICK_SHORT, MODE_ON):
-      if (SaberBase::IsOn()) { // prevents triggering during preon
-        if (SaberBase::GetColorChangeMode() != SaberBase::COLOR_CHANGE_MODE_NONE) {
-          ResetColorChangeMode();
+        current_menu_angle_ = (fusor.angle2() + (M_PI / 6));
+        return false;
+
+      case EVENTID(BUTTON_NONE, EVENT_TWIST_LEFT, MODE_ON | BUTTON_AUX):
+        if (!scroll_presets_) {
+          PVLOG_NORMAL << "** EFFECT_USER3 **\n";
+          SaberBase::DoEffect(EFFECT_USER3, 0);
           return true;
-        } else {
-          if (SFX_quote) {
-            if (sequential_quote_) {
-              SFX_quote.SelectNext();
-            } else {
-              SFX_quote.Select(-1);
-            }
-            SaberBase::DoEffect(EFFECT_QUOTE, 0);
-            // hybrid_font.PlayCommon(&SFX_quote);
-          } else {
-            SaberBase::DoForce();
-          }
         }
-      }
-      return true;
+        current_menu_angle_ = (fusor.angle2() - (M_PI / 6));
```

Comment:
This can make current_menu_angle_ less than -M_PI which might not be what you want.
Maybe use the new Angle class for current_menu_angle_ ?

Reply:
ah, common/angle.h?
ok so that basically auto wraps to keep it within pi to -pi range.
Would this be the correct syntax then?
`current_menu_angle_ = Angle(fusor.angle2()) - Angle(M_PI / 6);`

Reply:
That would work, yes. (Although the it also works without the second Angle(), as float will auto-cast to Angle in this case.)

Reply:
ok well I've just been messing with this saved twist stuff and the way I have it currently (just updated) seems to work as I intended.
Not resetting the angle after doing EFFECT_USERN, so this specific line goes away.

Reply:
Updated saved twist stuff.
Changed variable current_menu_angle to current_twist_angle,
Prevent normal twist after successful TWIST_LEFT/RIGHT gesture for USER effects.
Removed unused defines.